### PR TITLE
Publish wheels as well as an sdist

### PR DESCRIPTION
**Pull request by thejcannon**
This change changes CI so that wheels across most combinations of OS/Python are built and uploaded. It leverages https://github.com/pypa/cibuildwheel/tree/main to do all the heavy lifting.

Fixes https://github.com/HdrHistogram/HdrHistogram_py/issues/46

### CODE REVIEW

**Comment by thejcannon** [comment]
@ahothan Sorry to ping, but since I don't see a whole lot of activity, I wanted to toss this on your radar.

If you want I'd be happy to go so far as to try this workflow out on my fork, publishing to https://test.pypi.org/ (under a different name)

**Comment by thejcannon** [comment]
Ah, and I should mention, this takes advantage of the new shiny "Trusted Publisher" feature, you'll want to enable on PyPI. Then you can remove the `PYPI_API_TOKEN` secret from this repo.

> This requires setting this GitHub workflow in your project's PyPI settings (for a [new project](https://docs.pypi.org/trusted-publishers/creating-a-project-through-oidc)/[existing project](https://docs.pypi.org/trusted-publishers/adding-a-publisher)).

**Comment by ahothan** [comment]
> @ahothan Sorry to ping, but since I don't see a whole lot of activity, I wanted to toss this on your radar.
> 
> If you want I'd be happy to go so far as to try this workflow out on my fork, publishing to https://test.pypi.org/ (under a different name)

Thanks for the PR, I actually wanted to publish wheels but did not find a good way.
If you can test it out that would be great.
I have added a trusted publisher to PiPI and need to test it.

**Comment by thejcannon** [comment]
OK, I dirtied my fork's `master` branch with [this workflow file](https://github.com/thejcannon/HdrHistogram_py/actions/runs/5825831230/workflow) which ran [here](https://github.com/thejcannon/HdrHistogram_py/actions/runs/5825831230).

It made wheels, but died on upload due to the security handshaking.

Let me try again

**Comment by thejcannon** [comment]
I had the repo name wrong.

After fixing the trusted publisher with the correct repo name:

Screenshot from 2023-08-10 15-51-08

and re-running the failed job [here](https://github.com/thejcannon/HdrHistogram_py/actions/runs/5825831230/job/15798808220) it all worked!

See https://test.pypi.org/project/hdrhistogram2/#files

:tada:

**Comment by ahothan** [comment]
@thejcannon: cool, thanks for testing it!

**Review by ahothan — APPROVED** [review]

**Comment by thejcannon** [comment]
And to circle around: https://pypi.org/project/hdrhistogram/#files

Thanks you so much!

**Comment by ahothan** [comment]
thanks for the PR, surprising how many wheels have been built and some that are not present such as macos arm64.
I suppose we can skip some of the older options such as python 3.6, 3.7, ppc, ...

**Comment by thejcannon** [comment]
Yeah looks like `cibuildwheel` has many options: https://cibuildwheel.readthedocs.io/en/stable/options/ I just went with the defaults because, well, it's the defaults :sweat_smile:
